src/board.rs

```diff
@@ -106,123 +215,175 @@ impl Chessboard {
             ('b', self.black_bishops),
             ('k', self.black_king),
             ('q', self.black_queen),
-            ('r', self.black_rooks)
+            ('r', self.black_rooks),
         ]
     }
 
+    /// Foreground color to display for this piece
     /// # Return: The color of the piece
+    #[rustfmt::skip]
     fn find_fg(&self, p: char) -> Color {
-        if p.is_uppercase() { Color::White } else { Color::Black }
+        if p.is_uppercase() { Color::White }
+        else                { Color::Black }
     }
 
-
     /// # Return: The color of the board at this position
     fn find_bkgnd(&self, rank: usize, file: usize) -> Color {
         if (rank + file) % 2 == 0 {
-            return Color::Rgb { r: 190, g: 140, b: 170 };
+            return Color::Rgb {
+                r: 190,
+                g: 140,
+                b: 170,
+            };
         } else {
-            return Color::Rgb { r: 255, g: 206, b: 158 };
+            return Color::Rgb {
+                r: 255,
+                g: 206,
+                b: 158,
+            };
         }
     }
 
-
     /// Retrieve the chess piece at a specific position on the chessboard.
     /// * `rank` - The rank of the square.
     /// * `file` - The file (A=0) of the square.
     /// # Return:
     /// The character representation of the piece at this position.
     /// If there is no piece here it will return a period.
-    fn piece_at_position(&self, rank: usize, file: usize) -> char { 
+    fn piece_at_position(&self, rank: usize, file: usize) -> char {
         for (p_type, positions) in self.get_pieces() {
             let rank_byte = positions >> ((rank - 1) * 8);
-            if (rank_byte & (1 << file)) != 0 { return p_type; }
+            if (rank_byte & (1 << file)) != 0 {
+                return p_type;
+            }
         }
         '.'
     }
 
+    /// Forsyth–Edwards Notation Serializer
+    /// * `chessboard` - The chessboard position to be converted to a FEN.
+    /// # Return: FEN string representing the board's position.
+    pub fn to_string(&self) -> String {
```

Comment:
This function is wayyyy too long, we should break it down into smaller pieces for testing.